app/features-json/artifact_json_controller.rb

```diff
@@ -0,0 +1,79 @@
+require_relative "./json_authenticated_controller_base"
+require_relative "./view_models/artifact_view_model"
+
+module FastlaneCI
+  # Controller for providing all data relating to artifacts
+  class ArtifactJSONController < JSONAuthenticatedControllerBase
+    HOME = "/data/project/:project_id/build/:build_number/artifact"
+
+    class ArtifactNotFoundError < FastlaneCIError
+    end
+
+    # Fetch all the available metadata for a given artifact, the return data is JSON
+    get "#{HOME}/:artifact_id/info" do |project_id, build_number, artifact_id|
+      begin
+        artifact, = fetch_artifact_details(
```

Comment:
if destructuring and we have an un-used parameter, please use `_` instead of leaving it blank.